java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/RelationalSymbolicValue.java

```diff
@@ -172,35 +172,64 @@ RelationalSymbolicValue inverse() {
     return results;
   }
 
+  private boolean hasBooleanSymbolicValueAsOperand() {
+    return (leftOp instanceof BooleanSymbolicValue) || (rightOp instanceof BooleanSymbolicValue);
+  }
+
+  private boolean isEqualityCheck() {
+    return Kind.METHOD_EQUALS == kind || Kind.NOT_METHOD_EQUALS == kind || Kind.EQUAL == kind || Kind.NOT_EQUAL == kind;
```

Comment:
What about:
```
return isEquality() || Kind.NOT_METHOD_EQUALS == kind || Kind.NOT_EQUAL == kind;
```
In order to mark the difference with isEquality ?

Reply:
Nice idea!